sql/all_match_pairs_with_um.sql

```diff
@@ -1,44 +1,45 @@
 -- add "self matches" for the articles that didn't match anything (this can happen if the article has a lot of null
 -- fields) to the rest of the article match pairs
-WITH oa_matches AS (
-  SELECT
-    id AS id1,
-    ids.mag AS id2
-  FROM
-    openalex.works
-  WHERE
-    (ids.mag IS NOT NULL) AND ((type IS NULL)
-        OR NOT (type IN ("dataset", "peer-review", "grant")))
-  UNION ALL
-  SELECT
-    ids.mag AS id1,
-    id AS id2
-  FROM
-    openalex.works
-  WHERE
-    (ids.mag IS NOT NULL) AND ((type IS NULL)
-        OR NOT (type IN ("dataset", "peer-review", "grant")))
-  ),
-  s2_matches AS (
-    SELECT
-      CAST(corpusid AS string) as id1,
-      CAST(externalids.MAG AS string) as id2
+WITH lens_matches AS (
+    (SELECT
+      lens_id as id1,
+      CONCAT("https://openalex.org/", id.value) as id2
     FROM
-      semantic_scholar.papers
-    CROSS JOIN unnest(publicationtypes) as publication_type
+      lens.scholarly
+    CROSS JOIN
+      UNNEST(external_ids) as id
     WHERE
-      (externalids.MAG is not null) and NOT
-        (publication_type IN ("Dataset", "Editorial", "LettersAndComments", "News", "Review"))
+      (id.type = "openalex")
+      AND lens_id in (select id from {{ staging_dataset }}.lens_ids)
+    )
     UNION ALL
+    (
     SELECT
-      CAST(externalids.MAG AS string) as id1,
-      CAST(corpusid AS string) as id2
+      CONCAT("https://openalex.org/", id.value) as id1,
+      lens_id as id2
     FROM
-      semantic_scholar.papers
-    CROSS JOIN unnest(publicationtypes) as publication_type
+      lens.scholarly
+    CROSS JOIN
+      UNNEST(external_ids) as id
     WHERE
-      (externalids.MAG is not null) and NOT
-        (publication_type IN ("Dataset", "Editorial", "LettersAndComments", "News", "Review"))
+      (id.type = "openalex")
+      AND lens_id in (select id from {{ staging_dataset }}.lens_ids)
+    )
+    (
+    SELECT
```

Comment:
I'm guessing based on the name that the alias_lens_ids column maps lens ids to duplicates, so using those to match here

Reply:
`alias_lens_ids` is [undocumented](https://docs.api.lens.org/response-scholar.html) so I'm not sure either. But from brief inspection, seems likely. We always have 1+ alias, and from spot checking, the array of aliases includes the `lens_id`. 

```
select 
   array_length(alias_lens_ids) as n_aliases,
   count(*) n_pubs
from `gcp-cset-projects.lens.scholarly`
group by 1
order by 1
```